### DISCUSSION TITLE
Out of bounds range limit option

### DISCUSSION THREAD

**Discussion by michaelgeorge001** [jvde-github/AIS-catcher#90] [Ideas]
A feature that I would find useful would be the ability to drop entries based on out of bound ranges - like 181º longitude or -92º Latitude, or distances like 14097.2 km.

Maybe a range limit command switch like -R  LAT 90 LON 60 DIS 300, or something similar.

-Mike-

**Comment by jvde-github** [comment]
Hi Mike, what is your goal? To remove them from the webclient or from all output? The first is fairly easy to achieve, the second a bit more tricky as distances are calculated at webserver level and not available when sending out, e.g., UDP messages. 

Jasper

**Reply by michaelgeorge001** [reply]
Hi Jasper,

I'm sorry, I should have been more specific. My goal was to just remove them from the webclient because I've rarely noticed them on the services like Marine Traffic, VesselFinder, etc.

It's not a showstopper item for me but just one of those things that made me laugh. Like when I see a ship at 91ºN, or at Lat -80.55, Lon -139.04167 ( Continental Antartica) when I'm in Nova Scotia (tropospheric ducting aside).  I do have fun teasing my wife about how well my homemade AIS antenna is working out.

Mike

**Comment by jvde-github** [comment]
Great. One option (with the latest clone) is to use the below plugin that filters out ships > 1000 nmi away from the station. In practice I always rely on the validation flag. But not validated ships will still be shown on the map. 

We could refine this to use a shorter distance but still include ships further out if they have been validated as genuine by the engine (the shipcard and table have green indicators). 

https://github.com/jvde-github/AIS-Catcher-PLUGINS/blob/aee301e0052c8a25432d6f9791abd96b94310c58/filtership.pjs

Let me know if this works for you.
